lib/hydra/works/services/generic_file/upload_file.rb

```diff
@@ -6,11 +6,17 @@ def self.call(object, path, additional_services=[], replace=false)
       raise ArgumentError, "supplied path must be a string" unless path.is_a?(String)
       raise ArgumentError, "supplied path to file does not exist" unless ::File.exists?(path)
     
+      opts = { 
+        original_name: ::File.basename(path), 
+        mime_type: Hydra::Works::GetMimeTypeForFile.call(path),
+        replace: replace
+      }
+
       if object.original_file.nil?
-        Hydra::Works::AddOriginalFile.call(object, path)
+        Hydra::Works::AddOriginalFile.call(object, File.open(path), opts)
```

Comment:
After thinking about it for a few moments, I believe its' just the classes name that is proving confusing to me.  Something more like "UploadLocalFile" would be more clear to me.